### PR TITLE
WIP on 2.5.9 annoucement

### PR DESCRIPTION
**Pull request by ktoso**

### CODE REVIEW

File: blog/_posts/2018-01-11-akka-2.5.9-released.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+A total of 40 issues were closed since 2.5.8.
+The complete list can be found on the [2.5.9](https://github.com/akka/akka/milestone/125?closed=1) milestone on github.
+
+## Credits
+
+For this release we had the help of 24 committers – thank you all very much!
+
+```
+TODO RE GENERATE THESE
```
Comment (ktoso) [inline blog/_posts/2018-01-11-akka-2.5.9-released.md]: TODO change once final things merged

File: blog/_posts/2018-01-11-akka-2.5.9-released.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+title: Akka 2.5.9 Released
+author: Konrad 'ktoso' Malawski
+short: We are happy to announce a the 2.5.9 patch release of Akka 2.5
```
Comment (patriknw) [inline blog/_posts/2018-01-11-akka-2.5.9-released.md]: a the

File: blog/_posts/2018-01-11-akka-2.5.9-released.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+We are pleased to announce a new patch release of Akka 2.5. This release mostly contains small fixes across the board
+and our continued work on Akka Typed. Some notable improvements and bug fixes in 2.5.8 are:
+
+- Provide a fail-fast on version number too low for libraries [#24030](https://github.com/akka/akka/issues/24030)
+- expectNoMessage should consume the message from the queue [#24270](https://github.com/akka/akka/issues/24270)
+- DDataShardCoordinator: Provide read-only access when waitingForUpdate [#24064](https://github.com/akka/akka/issues/24064)
+- Graceful handover when cluster client receptionist shuts down [#24179](https://github.com/akka/akka/issues/24179)
+- Add overload with Function0 predicate to ReceiveBuilder [#15446](https://github.com/akka/akka/issues/15446)
+
```
Comment (patriknw) [inline blog/_posts/2018-01-11-akka-2.5.9-released.md]: ## Akka Typed

There are several breaking changes of Akka Typed.

Our goal is to ship a production ready version of Akka Typed as soon as possible and therefore we have removed the new runtime implementation and initially only support running typed actors with the underlaying untyped actor system. See the [Coexistence blog post]( https://akka.io/blog/2017/05/06/typed-coexistence).

Some major structural changes to Akka Typed have been done, such as splitting up in separate modules (e.g. `akka-actor-typed` with package root `akka.actor.typed`, and `akka-cluster-typed` with package `akka.cluster.typed`.

Akka Typed in 2.5.9 is not ready and we will continue working on it the next few weeks with more expected breaking changes. If you prefer a single migration task you should stay on version 2.5.8 until we completed these changes.

File: blog/_posts/2018-01-11-akka-2.5.9-released.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+A total of 40 issues were closed since 2.5.8.
+The complete list can be found on the [2.5.9](https://github.com/akka/akka/milestone/125?closed=1) milestone on github.
+
```
Comment (patriknw) [inline blog/_posts/2018-01-11-akka-2.5.9-released.md]: ## Akka 2.4.x End Of Life

Please be advised that Akka 2.4.x has reached its End Of Life (EOL) date as of 12/31/17. This especially means that guaranteed hotfixes for this branch will not be available after this point. We always recommend staying up to date with the latest release, which is currently Akka 2.5.9. When upgrading please make sure to utilize the [migration documentation](https://doc.akka.io/docs/akka/current/project/migration-guide-2.4.x-2.5.x.html) as necessary.

File: blog/_posts/2018-01-11-akka-2.5.9-released.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+and our continued work on Akka Typed. Some notable improvements and bug fixes in 2.5.8 are:
+
+- Provide a fail-fast on version number too low for libraries [#24030](https://github.com/akka/akka/issues/24030)
+- expectNoMessage should consume the message from the queue [#24270](https://github.com/akka/akka/issues/24270)
+- DDataShardCoordinator: Provide read-only access when waitingForUpdate [#24064](https://github.com/akka/akka/issues/24064)
+- Graceful handover when cluster client receptionist shuts down [#24179](https://github.com/akka/akka/issues/24179)
+- Add overload with Function0 predicate to ReceiveBuilder [#15446](https://github.com/akka/akka/issues/15446)
+
+A total of 40 issues were closed since 2.5.8.
+The complete list can be found on the [2.5.9](https://github.com/akka/akka/milestone/125?closed=1) milestone on github.
```
Comment (patriknw) [inline blog/_posts/2018-01-11-akka-2.5.9-released.md]: move this before the Typed section (I commented on wrong line)

File: blog/_posts/2018-01-11-akka-2.5.9-released.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+---
+
+Dear hakkers,
+
+We are pleased to announce a new patch release of Akka 2.5. This release mostly contains small fixes across the board
+and our continued work on Akka Typed. Some notable improvements and bug fixes in 2.5.8 are:
+
+- Provide a fail-fast on version number too low for libraries [#24030](https://github.com/akka/akka/issues/24030)
+- expectNoMessage should consume the message from the queue [#24270](https://github.com/akka/akka/issues/24270)
+- DDataShardCoordinator: Provide read-only access when waitingForUpdate [#24064](https://github.com/akka/akka/issues/24064)
```
Comment (patriknw) [inline blog/_posts/2018-01-11-akka-2.5.9-released.md]: change this to

* Optimizations of Cluster Sharding [#24064](https://github.com/akka/akka/issues/24270) and [#24191](https://github.com/akka/akka/issues/24191)

File: blog/_posts/2018-01-11-akka-2.5.9-released.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+tags: [releases]
+---
+
+Dear hakkers,
+
+We are pleased to announce a new patch release of Akka 2.5. This release mostly contains small fixes across the board
+and our continued work on Akka Typed. Some notable improvements and bug fixes in 2.5.8 are:
+
+- Provide a fail-fast on version number too low for libraries [#24030](https://github.com/akka/akka/issues/24030)
+- expectNoMessage should consume the message from the queue [#24270](https://github.com/akka/akka/issues/24270)
```
Comment (patriknw) [inline blog/_posts/2018-01-11-akka-2.5.9-released.md]: don't think this is worth mentioning

File: blog/_posts/2018-01-11-akka-2.5.9-released.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+- DDataShardCoordinator: Provide read-only access when waitingForUpdate [#24064](https://github.com/akka/akka/issues/24064)
+- Graceful handover when cluster client receptionist shuts down [#24179](https://github.com/akka/akka/issues/24179)
+- Add overload with Function0 predicate to ReceiveBuilder [#15446](https://github.com/akka/akka/issues/15446)
```
Comment (patriknw) [inline blog/_posts/2018-01-11-akka-2.5.9-released.md]: * New serializer for `Address` [#24265](https://github.com/akka/akka/issues/24265)
* Update to Aeron 1.7.0 for Artery [#24169](https://github.com/akka/akka/issues/24169)

**Comment by ktoso** [comment]
Thanks for the great additions!

File: blog/_posts/2018-01-11-akka-2.5.9-released.md (hunk 4 above)
Comment (ktoso) [reply blog/_posts/2018-01-11-akka-2.5.9-released.md]: Ah right, now is the time to mention it I guess. Adding, thanks a ton!

File: blog/_posts/2018-01-11-akka-2.5.9-released.md (hunk 3 above)
Comment (ktoso) [reply blog/_posts/2018-01-11-akka-2.5.9-released.md]: Also added 

> Upgrading to Akka 2.5.x from the 2.4.x series should be very simple, as 2.5 keeps backwards binary compatibility,
> in accordance to our [documented compatibility guidelines](http://doc.akka.io/docs/akka/current/scala/common/binary-compatibility-rules.html).

File: blog/_posts/2018-01-11-akka-2.5.9-released.md (hunk 4 above)
Comment (ktoso) [reply blog/_posts/2018-01-11-akka-2.5.9-released.md]: Also added 

> Upgrading to Akka 2.5.x from the 2.4.x series should be very simple, as 2.5 keeps backwards binary compatibility,
> in accordance to our [documented compatibility guidelines](http://doc.akka.io/docs/akka/current/scala/common/binary-compatibility-rules.html).

File: blog/_posts/2018-01-11-akka-2.5.9-released.md (hunk 7 above)
Comment (ktoso) [reply blog/_posts/2018-01-11-akka-2.5.9-released.md]: ok :)

File: blog/_posts/2018-01-11-akka-2.5.9-released.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
+---
+layout: post
+title: Akka 2.5.9 Released
```
Comment (patriknw) [inline blog/_posts/2018-01-11-akka-2.5.9-released.md]: should we mention 2.4.x EOL in title?

Comment (ktoso) [reply blog/_posts/2018-01-11-akka-2.5.9-released.md]: Good point, and filename

File: blog/_posts/2018-01-11-akka-2.5.9-released-2.4.x-end-of-life.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
+---
+layout: post
+title: Akka 2.5.9 Released and 2.4.x End-Of-Life
```
Comment (ktoso) [inline blog/_posts/2018-01-11-akka-2.5.9-released-2.4.x-end-of-life.md]: sounds good @patriknw ?

File: blog/_posts/2018-01-11-akka-2.5.9-released.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+## Akka Typed
+
+There are several breaking changes of Akka Typed.
+
+Our goal is to ship a production ready version of Akka Typed as soon as possible and therefore we have removed the new
+runtime implementation and initially only support running typed actors with the underlaying untyped actor system.
+See the [Coexistence blog post](https://akka.io/blog/2017/05/06/typed-coexistence).
+
+Some major structural changes to Akka Typed have been done, such as splitting up in separate modules
+(e.g. `akka-actor-typed` with package root `akka.actor.typed`, and `akka-cluster-typed` with package `akka.cluster.typed`.
```
Comment (patriknw) [inline blog/_posts/2018-01-11-akka-2.5.9-released.md]: looks like I missed end `)`, but perhaps remove the `(` instead

File: blog/_posts/2018-01-11-akka-2.5.9-released-2.4.x-end-of-life.md (hunk 10 above)
Comment (patriknw) [reply blog/_posts/2018-01-11-akka-2.5.9-released-2.4.x-end-of-life.md]: +1

File: blog/_posts/2018-01-11-akka-2.5.9-released.md (hunk 11 above)
Comment (ktoso) [reply blog/_posts/2018-01-11-akka-2.5.9-released.md]: Removed the `(`

File: blog/_posts/2018-01-11-akka-2.5.9-released-2.4.x-end-of-life.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+author: Konrad 'ktoso' Malawski
+short: We are happy to announce the 2.5.9 patch release of Akka 2.5
+category: news
+tags: [releases]
+---
+
+Dear hakkers,
+
+We are pleased to announce a new patch release of Akka 2.5. This release mostly contains small fixes across the board
+and our continued work on Akka Typed. Some notable improvements and bug fixes in 2.5.8 are:
```
Comment (johanandren) [inline blog/_posts/2018-01-11-akka-2.5.9-released-2.4.x-end-of-life.md]: in 2.5.9

Comment (ktoso) [reply blog/_posts/2018-01-11-akka-2.5.9-released-2.4.x-end-of-life.md]: Thanks :)

File: blog/_posts/2018-01-11-akka-2.5.9-released-2.4.x-end-of-life.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+## Akka 2.4.x End Of Life
+
+Please be advised that Akka 2.4.x has reached its End Of Life (EOL) date as of 12/31/2017.
+
+Upgrading to Akka 2.5.x from the 2.4.x series should be very simple, as 2.5 keeps backwards binary compatibility,
+in accordance to our [documented compatibility guidelines](http://doc.akka.io/docs/akka/current/scala/common/binary-compatibility-rules.html).
+
+This especially means that guaranteed hotfixes for this branch will not be available after this point.
+We always recommend staying up to date with the latest release, which is currently Akka 2.5.9.
+When upgrading please make sure to utilize the [migration documentation](https://doc.akka.io/docs/akka/current/project/migration-guide-2.4.x-2.5.x.html) as necessary.
```
Comment (raboof) [inline blog/_posts/2018-01-11-akka-2.5.9-released-2.4.x-end-of-life.md]: 👍

File: blog/_posts/2018-01-11-akka-2.5.9-released-2.4.x-end-of-life.md [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+Akka Typed in 2.5.9 is not ready yet, and we will continue working on it the next few weeks with more expected breaking changes.
+If you are using Typed already and would prefer a single migration task you should stay on version 2.5.8 until we have 
+completed these changes.
+
+## Credits
+
+For this release we had the help of 24 committers – thank you all very much!
+
+```
+TODO RE GENERATE THESE
```
Comment (raboof) [inline blog/_posts/2018-01-11-akka-2.5.9-released-2.4.x-end-of-life.md]: don't forget ;)

**Pull request merged**